plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiModule.java

```diff
@@ -76,7 +80,27 @@ public void configure(Binder binder)
     @Provides
     public ExecutorService createExecutorService()
     {
-        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%d"));
+        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%s"));
+    }
+
+    @ForHudiSplitSource
```

Comment:
Put these in the order
```java
@Provides
@Singleton
@ForHudiSplitSource
```
Then it reads naturally as "provides singleton for Hudi split source"